addons/sale_timesheet/views/hr_timesheet_views.xml

```diff
@@ -7,23 +7,89 @@
             <field name="inherit_id" ref="hr_timesheet.hr_timesheet_line_search"/>
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='department_id']" position="after">
-                    <field name="timesheet_invoice_type"/>
+                    <field name="so_line"/>
                 </xpath>
                 <xpath expr="//filter[@name='month']" position="before">
-                    <filter name="billable_timesheet" string="Billable" domain="[('so_line', '!=', False)]"/>
-                    <filter name="non_billable_timesheet" string="Non Billable" domain="[('so_line', '=', False)]"/>
-                    <separator/>
                     <filter name="billable_time" string="Billed on Timesheets" domain="[('timesheet_invoice_type', '=', 'billable_time')]"/>
                     <filter name="billable_fixed" string="Billed at a Fixed Price" domain="[('timesheet_invoice_type', '=', 'billable_fixed')]"/>
                     <filter name="non_billable" string="Non Billable Tasks" domain="[('timesheet_invoice_type', '=', 'non_billable')]"/>
                     <separator/>
                 </xpath>
                 <xpath expr="//filter[@name='groupby_department']" position="before">
-                        <filter string="Billable Type" name="groupby_timesheet_invoice_type" domain="[]" context="{'group_by': 'timesheet_invoice_type'}"/>
+                    <filter string="Billable Type" name="groupby_timesheet_invoice_type" domain="[]" context="{'group_by': 'timesheet_invoice_type'}"/>
+                </xpath>
+                <xpath expr="//filter[@name='groupby_employee']" position="after">
+                    <filter string="Sales Order Item" name="groupby_sale_order_item" domain="[]" context="{'group_by': 'so_line'}"/>
+                    <filter string="Invoice" name="groupby_invoice" domain="[]" context="{'group_by': 'timesheet_invoice_id'}"/>
                 </xpath>
             </field>
     </record>
 
+    <record id="hr_timesheet_line_tree_inherit" model="ir.ui.view">
+        <field name="name">account.analytic.line.tree.inherit</field>
+        <field name="model">account.analytic.line</field>
+        <field name="inherit_id" ref="hr_timesheet.hr_timesheet_line_tree"/>
+        <field name="arch" type="xml">
+            <xpath expr="//tree/field[@name='unit_amount']" position="before">
+                <field name="so_line" optional="hide" options="{'no_open': True}" readonly="1" context="{'create': False, 'edit': False, 'delete': False}" domain="[('is_service', '=', True), ('is_expense', '=', False), ('state', 'in', ['sale', 'done'])]"/>
```

Comment:
With this domain, the user could select a SOL for another customer, it is not good, if the task is for a "Customer A" and the user selects a SOL for "Customer B".
I think we must to create a field in `account.analytic.line` model to have the commercial partner based on the project, task or current SOL in the timesheet.
Moreover, we could add the domain the field definition instead of on the field in each view.

Reply:
Done.